thousandeyes/resource_stream.go

```diff
@@ -0,0 +1,217 @@
+package thousandeyes
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/thousandeyes/thousandeyes-sdk-go/v2"
+)
+
+type Stream struct {
+	ID                string            `json:"id,omitempty"`
+	Enabled           bool              `json:"enabled,omitempty"`
+	Type              string            `json:"type,omitempty"`
+	EndpointType      string            `json:"endpointType,omitempty"`
+	StreamEndpointUrl string            `json:"streamEndpointUrl,omitempty"`
+	DataModelVersion  string            `json:"dataModelVersion,omitempty"`
+	TestMatch         []StreamTestMatch `json:"testMatch,omitempty"`
+	TagMatch          []StreamTagMatch  `json:"tagMatch,omitempty"`
+}
+
+type StreamTestMatch struct {
+	ID     string `json:"id,omitempty"`
+	Domain string `json:"domain,omitempty"`
+}
+
+type StreamTagMatch struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+// StreamClient extends thousandeyes.Client such that we can use the (at the
+// time of writing) not-yet-implemented ThousandEyes v7 API.
+type StreamClient struct {
+	c *thousandeyes.Client
+}
+
+// NewStreamClientFrom copies an existing v6 client, modifies the URL in its
+// APIEndpoint field to point to v7, and returns a StreamClient with the
+// modified copy.
+func NewStreamClientFrom(v6 *thousandeyes.Client) *StreamClient {
+	v7 := new(thousandeyes.Client)
+	*v7 = *v6
+	v7.APIEndpoint = strings.ReplaceAll(v6.APIEndpoint, "v6", "v7")
+	return &StreamClient{
+		c: v7,
+	}
+}
+
+// Exposing a simpler version of do() because thousandeyes.Client won't.
+func (sc *StreamClient) do(method string, path string, payload interface{}) (*http.Response, error) {
+	if sc.c.Limiter != nil {
+		sc.c.Limiter.Wait()
+	}
+	endpoint := sc.c.APIEndpoint + path + ".json"
```

Comment:
We won't want the `.json` for v7
```suggestion
	endpoint := sc.c.APIEndpoint + path
```